assets/js/components/Header.js

```diff
@@ -48,12 +48,19 @@ const Header = ( { children, subHeader, showNavigation } ) => {
 		select( CORE_USER ).isAuthenticated()
 	);
 	const { y } = useWindowScroll();
+	const [ hasSubheader, setHasSubheader ] = useState( false );
+
+	const subHeaderElementRef = useCallback( ( node ) => {
+		setTimeout( () => {
+			setHasSubheader( node?.childNodes.length );
+		}, 1 ); // technically should be 0 but for failing Jest tests.
+	}, [] );
```

Comment:
After thinking about this a bit more and doing some testing, I found that this approach is a bit too simple to cover our needs as we can't rely on the child nodes being there as soon as the ref is set, or even 1sec later. Some may come from remote requests, and it's possible that the notification may appear for a split second only to be hidden/removed.

The key point here is that we need a solution that will update when children are added and removed, whenever that happens, not just once.

I found a solution that seems to work well using mutation observer and added a few tests, so I'll put together a quick PR for your review on top of this one.

Reply:
👍